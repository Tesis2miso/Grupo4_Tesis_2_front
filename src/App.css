.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.ddlLanguage {
  position: fixed;
  top: 20px; /* change this value to adjust the distance from the top */
  right: 20px;
  z-index: 9999;
  background-color: white;
  border-radius: 50%;
  width: 60px;
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.15);
}

.ddlLanguage:hover {
  cursor: pointer;
}