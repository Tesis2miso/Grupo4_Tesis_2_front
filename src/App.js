import { BrowserRouter, Route, Routes } from "react-router-dom";
import { Suspense } from "react";
import Login from "./components/login/Login";
import Signup from "./components/signup/Signup";
import Agenda from "./components/agenda/Agenda";
import Home from "./components/home/Home";
import useToken from "./components/utils/useToken";
import "./App.css";
import ProtectedRoute from "./components/utils/ProtectedRoute";
import DdlLanguage from "./components/utils/DdlLanguage";
import i18n from "./components/utils/i18n";
import Loading from "./components/utils/Loading";
import LocaleContext from "./components/utils/LocaleContext";
import { useState } from "react";

function App() {
  const [locale, setLocale] = useState(i18n.language);
  i18n.on("languageChanged", (lng) => setLocale(i18n.language));

  const { token, removeToken, setToken } = useToken();
  const loggedIn = () => {
    return !(!token && token !== "" && token !== undefined);
  };

  return (
    <div className="App">
      <LocaleContext.Provider value={{ locale, setLocale }}>
        <Suspense fallback={<Loading />}>
          <div className="ddlLanguage">
            <DdlLanguage />
          </div>
          <BrowserRouter>
            <Routes>
              <Route
                exact
                path="/signup"
                element={<Signup loggedIn={loggedIn()} setToken={setToken} />}
              ></Route>
              <Route
                path="/login"
                element={<Login loggedIn={loggedIn()} setToken={setToken} />}
              ></Route>
              <Route
                exact
                path="/"
                element={
                  <ProtectedRoute loggedIn={loggedIn()}>
                    <Home removeToken={removeToken} />
                  </ProtectedRoute>
                }
              />
              <Route
                exact
                path="/agenda"
                element={
                  <ProtectedRoute loggedIn={loggedIn()}>
                    <Agenda removeToken={removeToken} />
                  </ProtectedRoute>
                }
              />
            </Routes>
          </BrowserRouter>
        </Suspense>
      </LocaleContext.Provider>
    </div>
  );
}

export default App;
