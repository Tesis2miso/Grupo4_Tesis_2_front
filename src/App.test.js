import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import App  from './App';

describe('Tests_App', () => {
  let wrapper = shallow(<App />);

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  test('show_App', () => {
    expect(wrapper).toMatchSnapshot();
  });


})