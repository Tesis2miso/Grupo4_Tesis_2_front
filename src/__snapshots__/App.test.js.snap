// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests_App show_App 1`] = `
<div
  className="App"
>
  <ContextProvider
    value={
      Object {
        "locale": undefined,
        "setLocale": [Function],
      }
    }
  >
    <Suspense
      fallback={<Loading />}
    >
      <div
        className="ddlLanguage"
      >
        <DdlLanguage />
      </div>
      <BrowserRouter>
        <Routes>
          <Route
            element={
              <Signup
                loggedIn={false}
                setToken={[Function]}
              />
            }
            exact={true}
            path="/signup"
          />
          <Route
            element={
              <Login
                loggedIn={false}
                setToken={[Function]}
              />
            }
            path="/login"
          />
          <Route
            element={
              <ProtectedRoute
                loggedIn={false}
              >
                <Home
                  removeToken={[Function]}
                />
              </ProtectedRoute>
            }
            exact={true}
            path="/"
          />
        </Routes>
      </BrowserRouter>
    </Suspense>
  </ContextProvider>
</div>
`;
